src/popup/components/TransactionList.vue

```diff
@@ -50,179 +40,230 @@ import {
+    }
 
-    const showSearchAndFilters = computed(() => (
-      props.showFilters
-      || displayMode.value.key !== FILTER_MODE.all
-      || searchPhrase.value
-    ));
-
-    function checkLoadMore() {
-      if (isDestroyed.value || !transactions.value.nextPageUrl) return;
-      // TODO - use viewport.ts composable after rewriting component to Vue 3
-      const isDesktop = document.documentElement.clientWidth > MOBILE_WIDTH
-          || process.env.IS_EXTENSION;
-      const { scrollHeight, scrollTop, clientHeight } = (isDesktop
-        ? document.querySelector('.app-inner') : document.documentElement)!;
-      if (props.maxLength && filteredTransactions.value.length >= props.maxLength) return;
-      if (scrollHeight - scrollTop <= clientHeight + 100) {
-        // eslint-disable-next-line no-use-before-define
-        loadMore();
+    function narrowTransactionsToDefinedToken(transactionList: ICommonTransaction[]) {
+      if (props.token) {
+        return transactionList.filter((transaction) => {
+          const innerTx = getInnerTransaction(transaction.tx);
+          if (props.token !== AETERNITY_CONTRACT_ID) {
+            return innerTx?.contractId === props.token;
+          }
+
+          return !innerTx.contractId || !isFungibleTokenTx(innerTx);
+        });
       }
+      return transactionList;
     }
 
-    async function loadMore() {
-      if (loading.value) return;
+    function filterTransactionsByDisplayMode(transactionList: ICommonTransaction[]) {
+      return transactionList.filter((transaction) => {
+        const outerTx = transaction.tx!;
+        const innerTx = transaction.tx ? getInnerTransaction(transaction.tx) : null;
+
+        const txOwnerAddress = getTxOwnerAddress(innerTx);
+
+        const direction = getTxDirection.value(
+          outerTx.payerId ? outerTx : innerTx,
+          (transaction as ITransaction).transactionOwner
+          || (
+            getOwnershipStatus(activeAccount.value, accounts.value, innerTx)
+            !== TRANSACTION_OWNERSHIP_STATUS.current
+            && txOwnerAddress
+          ),
+        );
+
+        const isDex = isTxDex(innerTx, getDexContracts.value);
+
+        switch (displayMode.value.key) {
+          case FILTER_MODE.all:
+            return true;
+          case FILTER_MODE.dex:
+            return isDex;
+          case FILTER_MODE.out:
+            return direction === TX_DIRECTION.sent && !isDex;
+          case FILTER_MODE.in:
+            return direction === TX_DIRECTION.received;
+          default:
+            throw new Error(`${i18n.t('pages.recentTransactions.unknownMode')} ${displayMode.value.key}`);
+        }
+      });
+    }
+
+    function filterTransactionsBySearchPhrase(transactionList: ICommonTransaction[]) {
+      return transactionList.filter(
+        (transaction) => !searchPhrase.value || getTxSymbol.value(transaction)
+          .toLocaleLowerCase()
+          .includes(searchPhrase.value.toLocaleLowerCase()),
+      );
```

Comment:
If I'm not mistaken we have a function to compare lower case strings

Reply:
no, we have utils to check equals, not for includes